File: flipperf/src/main/java/com/flipkart/flipperf/newlib/response/ResponseHandler.java
package com.flipkart.flipperf.newlib.response;

/**
 * Implementations are to read the number of bytes in case the response header do not have content-length
 */
public interface ResponseHandler {

    void onRead(int numBytes);

    /**
     * Signals that EOF has been reached reading the response stream from the network
     * stack.
     */
    void onEOF();
}

File: library-kotlin/src/main/java/com/flipkart/okhttpstats/kotlin/response/ResponseHandler.kt
package com.flipkart.okhttpstats.kotlin.response

/**
 * Implementations are to read the number of bytes in case the response header do not have content-length
 */
interface ResponseHandler {

    fun onRead(numBytes: Int)

    /**
     * Signals that EOF has been reached reading the response stream from the network
     * stack.
     */
    fun onEOF()
}